solidity/contracts/l2/L2TBTC.sol

```diff
@@ -0,0 +1,81 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/ERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// TODO:
+// * Be paused by any one of `n` guardians, allowing avoidance of contagion in case
+//  of a chain- or bridge-specific incident.
+// * Mint by minter.
+// * Proper documentation.
+// * Test initialization
+contract L2TBTC is ERC20Upgradeable, OwnableUpgradeable {
+    /// @notice Indicates if the given address is a Minter. Only Minters can
+    ///         mint the token.
+    mapping(address => bool) public isMinter;
+
+    /// @notice List of all Minters.
+    address[] public minters;
+
+    event MinterAdded(address indexed minter);
+    event MinterRemoved(address indexed minter);
+
+    modifier onlyMinter() {
+        require(isMinter[msg.sender], "Caller is not a minter");
+        _;
+    }
+
+    function initialize(string memory _name, string memory _symbol)
+        external
+        initializer
+    {
+        __Ownable_init();
+        __ERC20_init(_name, _symbol);
+    }
+
+    /// @notice Adds the address to the Minter list.
+    function addMinter(address minter) external onlyOwner {
+        require(!isMinter[minter], "This address is already a minter");
+        isMinter[minter] = true;
+        minters.push(minter);
+        emit MinterAdded(minter);
+    }
+
+    /// @notice Removes the address from the Minter list.
+    function removeMinter(address minter) external onlyOwner {
+        require(isMinter[minter], "This address is not a minter");
+        delete isMinter[minter];
+
+        // We do not expect too many Minters so a simple loop is safe.
+        for (uint256 i = 0; i < minters.length; i++) {
```

Comment:
The alternative might be that we reuse `isMinter` map and make it e.g. `mapping(address => index)` where `index` denotes an index in the `minters` array of a given address. This way we would get rid of the loop. We can check what is the gas implication for it. Probably `isMinter` should be named a bit differently in this case. `minters[0]` should also be probably reserved for things like `isMinter[nonExistantMinterAddr]`

```
        uint256 index = isMinter[minter]
        address minter = minters[minters.length - 1]
        address minterToRemove = minters[index]
        minters[index] = minter;
        minters.pop()
        isMinter[minter] = index
        delete isMinter[minterToRemove];
```

Reply:
The issue with the `0` index minter is pretty rough. Sticking with this construction also means an idiomatic public ABI `function isMinter(address m) view returns (bool);`

Reply:
I checked the gas implication and it seems that for a shorter set of minters, the gas usage is pretty similar.

Reply:
Are we good to resolve this conversation then?